packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-ciemss-node.vue

```diff
@@ -181,6 +183,21 @@ watch(
 			state.inProgressPostForecastId = postForecastId;
 			state.optimizedInterventionPolicyId = newInterventionResponse.id ?? '';
 			emit('update-state', state);
+		} else {
+			// Poller did not complete successfully
+			const simulation = await getSimulation(optId);
+			const state = _.cloneDeep(props.node.state);
+			if (simulation?.status && simulation?.statusMessage) {
+				state.optimizeErrorMessage = {
+					name: optId,
+					value: simulation.status,
+					traceback: simulation.statusMessage
+				};
+			}
```

Comment:
Assuming you are calling `pollAction` somewhere deep, then it has some basic error handling, and it might be better to extend that part rather than everyone doing their own thing. See:

https://github.com/DARPA-ASKEM/terarium/blob/25d4855995039fac8267e089b7050505f17eb014/packages/client/hmi-client/src/services/models/simulation-service.ts#L350-L353

Also the comment is a bit misleading, it isn't that poller did not complete successfully. It did but the simulation errored out.  If you want to handle "poller errors" you might need to wrap the polling parts inside try/catch blocks.

Reply:
Thanks Ill take a deeper look